src/software/ai/hl/stp/tactic/goalie/goalie_fsm.cpp

```diff
@@ -106,10 +107,42 @@ Point GoalieFSM::findGoodChipTarget(
     {
         chip_target = open_areas[0].origin();
     }
-
     return chip_target;
 }
 
+bool GoalieFSM::shouldEvacuateCrease(const Update &event)
+{
+    Rectangle friendly_defense_area = event.common.world.field().friendlyDefenseArea();
+    Ball ball                       = event.common.world.ball();
+
+    // calculate inflated crease obstacle
+    double robot_radius_expansion_amount =
+        ROBOT_MAX_RADIUS_METERS *
+        robot_navigation_obstacle_config.robot_obstacle_inflation_factor();
+    Rectangle inflated_defense_area =
+        friendly_defense_area.expand(robot_radius_expansion_amount);
+
+    bool ball_in_dead_zone =
+        !contains(friendly_defense_area, ball.position()) &&
+        contains(friendly_defense_area.expand(robot_radius_expansion_amount),
```

Comment:
I think you meant to use `inflated_defense_area` here instead. 

Since `inflated_defense_area` is a constant rectangle, and it's also generated in `shouldPanic` (though also not used?!) I think we should make it a field and move its creation to the constructor.

Reply:
you're right about the inflated defense area

aside: how could I access the equivalent of `event.common.world_ptr->field().friendlyDefenseArea()` from the constructor? I need it to make the inflated defense area

Reply:
That's a good point, I don't think you could do it from the constructor then. Though you could calculate `robot_radius_expansion_amount` in the constructor and store it as a constant field. 

Also, I think the changes suggested about the bug and the unused variable are still pending.

Reply:
Should be fixed now